pytest_logger/plugin.py

```diff
@@ -101,9 +103,11 @@ def pytest_runtest_makereport(self, item, call):
 
 
 class LoggerState(object):
-    def __init__(self, item, stdoutloggers, fileloggers):
+    def __init__(self, item, stdoutloggers, fileloggers, formatter=None):
```

Comment:
It might be easier to decide as early as possible which formatter we have, and pass it here as an existing member, instead of an optional one (by "optional" I mean: existing or `None`).

Reply:
Positional arg since 666aa66. Assuming it has been properly instantiated beforehand.